src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -298,22 +298,27 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		// Load the password if it's not already loaded
 		return this._connectionStore.addSavedPassword(connection).then(async result => {
 			let newConnection = result.profile;
-			let foundPassword = result.savedCred;
+			let hasSavedCred = result.savedCred;
 
 			// If there is no password, try to load it from an existing connection
-			if (!foundPassword && this._connectionStore.isPasswordRequired(newConnection)) {
+			if (!hasSavedCred && this._connectionStore.isPasswordRequired(newConnection)) {
 				let existingConnection = this._connectionStatusManager.findConnectionProfile(connection);
 				if (existingConnection && existingConnection.connectionProfile) {
 					newConnection.password = existingConnection.connectionProfile.password;
-					foundPassword = true;
+					hasSavedCred = true;
 				}
 			}
 
 			// Fill in the Azure account token if needed and open the connection dialog if it fails
 			let tokenFillSuccess = await this.fillInOrClearAzureToken(newConnection);
 
+			// If there's a token then there's a valid credential
+			if (tokenFillSuccess) {
```

Comment:
Can you explain why this is necessary? When would we have a token credential AND require a password such that this is needed?